### DISCUSSION TITLE
Need help with using stand alone flake in systemPackages

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2277] [Q&A] [answered]
I am trying to migrate to flakes for the past couple of days.
I have Nixvim as a stand-alone flake and running `nix run` works.
Here is my NixOS flake:
```nix
{
  description = "A very basic flake";

  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    nixvim = {
      url = "github:nix-community/nixvim";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs =
    { self, nixpkgs, ... }@inputs:
    let
      system = "x86_64-linux";
      pkgs = import nixpkgs {
        inherit system;
      };
    in
    {
      nixosConfigurations = {
        myNixos = nixpkgs.lib.nixosSystem {
          specialArgs = {
            inherit inputs system;
          };

          modules = [
            ./configuration.nix
            {
              environment.systemPackages =[
                inputs.nixvim.packages.${system}.default
              ];
            }
          ];
        };
      };
    };
}
```
I am trying to use the stand-alone Nixvim flake in my sytemPackages.
I get the following error when running `sudo nixos-rebuild switch --flake ~/repos/dotfiles/#myNixos`:
```
error: attribute 'default' missing
```

The stand-alone Nixvim flake defently contains a package called `defaul`t - I used the following template: `nix flake init --template github:nix-community/nixvim`.

Please help :-)

**Comment by MattSturgeon** [comment, marked as answer]
Your nixvim input is pointing to our flake, however it sounds like you have your own flake you wish to use.

Your error is telling you that `inputs.nixvim.packages.${system}.default` does not exist, which is correct, because `github:nix-community/nixvim` does not have a `packages.*.default` flake output.

You should be able to fix your issue by changing your nixvim input to point to _your_ new standalone nixvim flake, instead of ours.

As a sidenote: it is not required to use separate flakes in order to have a standalone nixvim build. Personally I have both a nixosConfiguration and a standalone nixvim package as two outputs on my flake, I also pass `self.packages.${system}.nvim` to my `environment.systemPackages` in my nixos configurations.

**Reply by refaelsh** [reply]
It worked! Thank you very much :-)

**Comment by refaelsh** [comment]
For future googlers, here is the final solution:
```nix
{
  description = "A very basic flake";

  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    nixvim.url = "./nixvim";
  };

  outputs =
    { self, nixpkgs, ... }@inputs:
    let
      system = "x86_64-linux";
      pkgs = import nixpkgs {
        inherit system;
      };
    in
    {
      nixosConfigurations = {
        myNixos = nixpkgs.lib.nixosSystem {
          specialArgs = {
            inherit inputs system;
          };

          modules = [
            ./configuration.nix
            {
              environment.systemPackages =[
                inputs.nixvim.packages.${system}.default
              ];
            }
          ];
        };
      };
    };
}
```
